hledger-lib/Hledger/Data/MarketPrice.hs

```diff
@@ -12,22 +12,35 @@ are thousands separated by comma, significant decimal places and so on.
 
 -}
 
-{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings, LambdaCase #-}
 
 module Hledger.Data.MarketPrice
 where
 import qualified Data.Text as T
--- import Test.HUnit
+import Test.HUnit
 
 import Hledger.Data.Amount
-import Hledger.Data.Types
 import Hledger.Data.Dates
+import Hledger.Data.Types
+-- import Hledger.Read.JournalReader
 import Hledger.Utils
+import Hledger.Utils.Parse
 
+-- | Get the string representation of an market price, based on its
+-- commodity's display settings.
 showMarketPrice :: MarketPrice -> String
 showMarketPrice mp = unwords
     [ "P"
     , showDate (mpdate mp)
     , T.unpack (mpcommodity mp)
     , (showAmount . setAmountPrecision maxprecision) (mpamount mp)
```

Comment:
It might be a bug with amount parsing then; I had a file with a currency given in small fractions of USD:

~~~
P 2016/07/01 VND $0.000045
~~~

And, without the precision forcing, it got printed out as

~~~
P 2016/07/01 VND $0.00
~~~

Might investigate this to see if the problem is actually the precision parsing.  I am also bothered by the precision forcing, which feels like an ugly hack that could break a lot of situations.